### PR TITLE
Refactor base facility kr

### PR DESCRIPTION
**Pull request by edmkitty**
I have refactored the base facility for enhanced readability. Nothing should have functionally changed.

### CODE REVIEW

**Comment by rtravitz** [comment]
FYI @edmkitty will be out until 8/22, so I'll take over shepherding the PR through review

**Comment by dhall** [comment]
Please don't read this as critical, but I'm curious why we're going this direction (pulling out of the base class) .. Do you feel the duplication makes it substantially more readable? The +/- on the changes looks show -205, + 318 so 50% more code. I don't think that less code is *always* better but generally consider it a win when we go that direction.

**Comment by nfasulo** [comment]
FYI this PR has been deprioritized for the moment.

**Comment by edmkitty** [comment]
@dhall I do think that pulling some of the methods out of the base class makes the code more readable, especially since some of the methods on the base class only pertained to a specific child class in some cases. For example, the `mh_clinic_phone`, `satisfaction_data`, `wait_time_data`, and `zip_plus_four` methods are only applicable to the vha_facility class. As for the duplication of the `pull_source_data` method, the reason I pulled that out into the child classes is because specifically in the case of the health facilities, we will be changing the data source entirely. So even though I have moved the health facility code into the vha_facility class, I think this will all be completely rewritten in the process of this epic: #2373.  If we decide to change the data sources for any other facility types, this change should help that along as well.

File: modules/va_facilities/lib/va_facilities/param_validators.rb [hunk 1]
```diff
@@ -30,8 +30,12 @@ def validate_no_services_without_type
...
       raise Common::Exceptions::InvalidFieldValue.new('type', params[:type]) unless
       BaseFacility::TYPES.include?(params[:type])
-      unknown = params[:services].to_a - BaseFacility::SERVICE_WHITELIST[params[:type]]
+      unknown = params[:services].to_a - facility_klass.service_list
       raise Common::Exceptions::InvalidFieldValue.new('services', unknown) unless unknown.empty?
     end
+
+    def facility_klass
+      BaseFacility::TYPE_MAP[params[:type]].constantize
+    end
```
[the remark this answers is not in the document]
Comment (edmkitty) [reply modules/va_facilities/lib/va_facilities/param_validators.rb]: Talked to @dvogel a bit, found that this test should cover that case: https://github.com/department-of-veterans-affairs/vets-api/blob/refactor_base_facility_kr/modules/va_facilities/spec/requests/facilities_request_spec.rb#L346

**Comment by dhall** [comment]
Thanks for the explanation @edmkitty ... I trust fresh eyes from you and your team regarding readability more than mine since I've spent a lot of time in this code :) If TeamZ is good with the changes, I have no objections 👍

**Pull request merged**